### ISSUE TITLE
Improve Prometheus labels from HAProxy Exporter

### ISSUE THREAD

**Issue by julianvmodesto** [appscode/voyager#271]
Currently, the backend label applied to the HAProxy exporter metrics with the HAProxy backend doesn't identify the Ingress.

```
haproxy_backend_current_session_rate{backend="http-service-zzt4r5"} 0
```

Could we change the name of the `backend` to be prefixed by the Ingress name?

For example, for an Ingress with `metadata.name = my-app`:

```
haproxy_backend_current_session_rate{backend="my-app-http-service-zzt4r5"} 0
```

This would allow for filtering by Ingress in Prometheus. Alternatively, a new label could be added, but I don't know if HAProxy Exporter or the Prometheus ServiceMonitor will allow adding arbitrary static labels.

**Comment by tamalsaha** [comment]
I think we can also add a label `{app="metadata.name"}`. Does that sound better?

**Comment by julianvmodesto** [comment]
Yes that would be perfect -- wasn't sure if this was possible.

**Comment by tamalsaha** [comment]
We are using a forked version of HAProxy exporter so that we can integrate this with Kube. https://github.com/appscode/haproxy_exporter . We can add extra labels, etc.

Is there anything else you want to see in the labels?

**Comment by julianvmodesto** [comment]
A label for the underlying Pod would be helpful as well.

**Comment by tamalsaha** [comment]
> A label for the underlying Pod would be helpful as well.

@julianvmodesto , Unfortunately we can't do this, since these label mapping rules are generated by Prometheus operator.

**Comment by julianvmodesto** [comment]
Hmm would it be feasible to rename the backend to be the pod name?

**Comment by tamalsaha** [comment]
> would it be feasible to rename the backend to be the pod name?

Can you show me some screenshot? I don't think I fully understand what you are referring to. Are you talking about the pod name in the labels? I see them.

![prometheus time series collection and processing server](https://user-images.githubusercontent.com/94814/28444454-2f9290bc-6d72-11e7-8e5b-c0b7ce21e5ff.png)

**Comment by tamalsaha** [comment]
I think I understand what you are referring to. You want to change the "backend" label. Currently this is generated in haproxy.config. So, I guess it is hard to follow.

If that is the issue, we are fixing that. #310 is essentially the same issue.

**Comment by julianvmodesto** [comment]
Yes that's right!
